src/edit_collector.h

```diff
@@ -17,30 +17,64 @@ class EditCollector {
     auto cf_id = edit.column_family_id_;
     auto& collector = column_families_[cf_id];
 
-    Status s;
     for (auto& file : edit.added_files_) {
-      s = collector.AddFile(file);
-      if (!s.ok()) return s;
+      status_ = collector.AddFile(file);
+      if (!status_.ok()) return status_;
     }
     for (auto& file : edit.deleted_files_) {
-      collector.DeleteFile(file.first, file.second);
-      if (!s.ok()) return s;
+      status_ = collector.DeleteFile(file.first, file.second);
+      if (!status_.ok()) return status_;
     }
 
     if (edit.has_next_file_number_) {
       if (edit.next_file_number_ < next_file_number_) {
-        return Status::Corruption("Edit has a smaller next file number " +
-                                  ToString(edit.next_file_number_) +
-                                  " than current " +
-                                  ToString(next_file_number_));
+        status_ =
+            Status::Corruption("Edit has a smaller next file number " +
+                               ToString(edit.next_file_number_) +
+                               " than current " + ToString(next_file_number_));
+        return status_;
       }
       next_file_number_ = edit.next_file_number_;
       has_next_file_number_ = true;
     }
     return Status::OK();
   }
 
+  Status Check(VersionSet& vset) {
+    if (!status_.ok()) {
+      return status_;
+    }
+
+    for (auto& cf : column_families_) {
+      auto cf_id = cf.first;
+      auto storage = vset.GetBlobStorage(cf_id).lock();
+      if (!storage) {
+        // TODO: support OpenForReadOnly which doesn't open DB with all column
+        // family so there are maybe some invalid column family, but we can't
+        // just skip it otherwise blob files of the non-open column families
+        // will be regarded as obsolete and deleted.
+        continue;
+      }
+      status_ = cf.second.Check(storage);
+      if (!status_.ok()) return status_;
+    }
+
+    checked_ = true;
+    return Status::OK();
+  }
+
   Status Apply(VersionSet& vset) {
+    if (!status_.ok()) {
+      return status_;
+    }
+
+    if (!checked_) {
+      status_ = Check(vset);
```

Comment:
The API is a bit confusing since caller can call `AddEdit` after `Check`, where we should double check for error. I think we can rename `Check` to `Seal` and return error if `AddEdit` is called after sealed.